### ISSUE TITLE
Unable to lint .eslintrc.js

### ISSUE THREAD

**Issue by michael-wolfenden** [eslint/eslint#6645]
**What version of ESLint are you using?**
3.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
module.exports = {
    "rules": {
        "semi": [
            "error",
            "never"
        ]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
The only files I have in my project is the configuration file with a semi-colon

```
// .eslintrc.js
module.exports = {
    "rules": {
        "semi": [
            "error",
            "never"
        ]
    }
};
```

and another file that also breaks the no-semi colon rule

```
// index.js
var test = "this should cause an eslint error";
```

I ran the following command 

`eslint .`

Expecting two errors (one for both files, but only got an error for `index.js`

```
C:\Dev\linting-eslintrc-issue\index.js
  1:47  error  Extra semicolon  semi

? 1 problem (1 error, 0 warnings)
```

I then ran

`eslint .eslintrc.js`

resulting in

```
C:\Dev\linting-eslintrc-issue\.eslintrc.js
  0:0  warning  File ignored by default.  Use a negated ignore pattern (like "--ignore-pattern '!<relative/path/to/filename>'") to override
```

So I assumed I had to un-ignore the eslintrc file, so I tried the following two commands

`eslint --ignore-pattern '!.eslintrc.js' .`
`eslint --ignore-pattern '!.\.eslintrc.js' .`

But still only saw an error results for `index.js`

```
C:\Dev\linting-eslintrc-issue\index.js
  1:47  error  Extra semicolon  semi

? 1 problem (1 error, 0 warnings)
```

**How can I get eslint to validate the current project directory as well as the `.eslintrc.js` file?**

**Comment by alberto** [comment]
Duplicate of https://github.com/eslint/eslint/issues/4828
